src/Type/Php/ArrayReverseFunctionReturnTypeExtension.php

```diff
@@ -43,6 +45,16 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 			return TypeCombinator::union(...$results);
 		}
 
+		if ($preserveKeys) {
+			// Remove list
+			$newArrayType = new ArrayType($type->getIterableKeyType(), $type->getIterableValueType());
+			if ($type->isIterableAtLeastOnce()->yes()) {
+				$newArrayType = TypeCombinator::intersect($newArrayType, new NonEmptyArrayType());
+			}
+
+			return $newArrayType;
```

Comment:
An almost identical construct exists already somewhere. And I might add it another time somewhere else. I was wondering if it makes sense to re-use this somehow somewhere..

Reply:
the cleaner alternative of course would be a refactor that moves `reverse()` to `Type`. but not sure if that's the way to go..

Reply:
yeah, moving this to type fixes like at least 3 other edge cases as well....